src/testModel.ts

```diff
@@ -551,6 +551,7 @@ export class TestModel extends DisposableBase {
       connectWsEndpoint: showBrowser ? externalOptions.connectWsEndpoint : undefined,
       updateSnapshots: noOverrideToUndefined(this._embedder.settingsModel.updateSnapshots.get()),
       updateSourceMethod: noOverrideToUndefined(this._embedder.settingsModel.updateSourceMethod.get()),
+      pageSnapshot: 'only-on-failure',
```

Comment:
if we release this is the default in `@playwright/test`, we can remove this override again.